src/chatd.cpp

```diff
@@ -1975,42 +1985,52 @@ bool Chat::setMessageSeen(Idx idx)
     }
 
     auto wptr = weakHandle();
-    karere::Id id = msg.id();
-    marshallCall([wptr, this, id, idx]()
+    megaHandle mEchoTimer = karere::setTimeout([this, wptr, idx, &msg, &mEchoTimer]()
     {
         if (wptr.deleted())
-            return;
-
-        CHATID_LOG_DEBUG("setMessageSeen: Setting last seen msgid to %s", ID_CSTR(id));
-        sendCommand(Command(OP_SEEN) + mChatId + id);
-        
-        Idx notifyStart;
-        if (mLastSeenIdx == CHATD_IDX_INVALID)
         {
-            notifyStart = lownum()-1;
+          return;
         }
-        else
+        mClient.mSeenTimers.erase(mEchoTimer);
+        mEchoTimer = 0;
+        karere::Id id = msg.id();
+        marshallCall([wptr, this, id, idx]()
```

Comment:
Theorically, no need the marshal anymore if we have a timer, since the lambda to be executed upon expiration of the timer will happen in the MEGAchat's thread, so the access to the socket is already enforced to be done in the MEGAchat's thread.